groupie/src/main/java/com/genius/groupie/Section.java

```diff
@@ -78,94 +125,128 @@ protected boolean isEmpty() {
         return children.isEmpty() || getItemCount(children) == 0;
     }
 
-    private void hideHeadersAndFooters() {
-        isHidden = true;
-        int count = getHeaderItemCount() + getFooterItemCount();
+    private void hideDecorations() {
+        int count = getHeaderItemCount() + getPlaceholderItemCount() + getFooterItemCount();
+        headerAndFooterHidden = true;
+        placeholderShown = false;
         notifyItemRangeRemoved(0, count);
     }
 
-    protected void showHeadersAndFootersIfNecessary() {
-        if (hideWhenEmpty && isHidden && !isEmpty()) {
-            showHeadersAndFooters();
+    private void configureEmptyState() {
+        boolean isEmpty = isEmpty();
+        if (isEmpty) {
+            if (hideWhenEmpty) {
+                if (!headerAndFooterHidden) {
```

Comment:
try and move logic into the hideDecorations() (and same for others like showPlaceholder)